Add consent decision-making process documentation (#887)
WordPress/openverse: documentation/reference/consent_decision_making.md

# Consent decision-making

> The Openverse maintainers have elected to adopt a decision-making model based on
> the

Reviewer: ```suggestion
The Openverse uses a decision-making model based on
the
```

[Sociocracy practice of "consent decision-making"](https://patterns.sociocracy30.org/consent-decision-making.html).
We hope this will accomplish the following:

- Bring expediency to decision-making by giving guidelines for how and what

>   types of feedback to give

Reviewer: Does "Bring expediency" here mean "speed up", as in "expedite", or "expediency" as in convenience/practicality/suitability? 

Not trying to be pedantic, but if speed is the goal a term like "speed up" might be clearer to contributors.

Author: Both but primarily the latter. Speed in and of itself does not seem, to me, a reasonable goal unless we just want to break things. Making expedient decisions, on the other hand, means we approach problem-solving and decision-making with more practicality to avoid getting bogged down in details that may not need to be addressed "_now_" (at the time of the particular decision). In particular this is achieved via the paramount/non-paramount objection distinction.

Please feel free to suggest alternative wording here. I can't think of anything that doesn't accidentally emphasis speed (e.g., "Expedite ...") or make this even wordier.

Reviewer: Expediency seems like the perfect word here, then.

> - Adopt more specific standards about what types of feedback must be addressed

Author: Both, assuming the standards are in the regular ongoing process of iteration and redefinition. This document proposes a few broad distinctions but also invites the team to change and add to those as needed to improve the process. I've made a change to clarify this. Let me know if you think further changes are necessary.

```suggestion
- Clarify the types of feedback expected during the decsion-making process, including which types of feedback must be addressed and which are marginal 
```

  before a proposal can be ratified for implementation
- Give specificity in feedback expectations, both for who will participate in a
  decision-making process and what responsibilities and tools they have for
  participating

> - Reaffirm trust between contributors when evaluating proposals

Reviewer: 👍

Historically, each of these have presented problems for the Openverse project as
it has grown. Decision-making can drag on for months, proposal reviews get stuck
in feedback cycles over issues that could be addressed later on, and there were
few, if any, expectations around who would participate in a discussion or what
the expectations of that participation was. Adopting the process described
below, we hope, will address some of the underlying causes of these problems, in

> addition to the benefits described above.

Reviewer: This is good background for the motivation to adopt this, of course, but in making the document as succinct as possible and "get to the point" as early as it can, would you consider removing it or adding it to some type of appendix towards the end? Maybe a section on history or motivation or something?

I think the focus of the document should be "here is how we communicate in Openverse" and provide the necessary background for participating as quickly as possible.

Author: Typically, sections on motivation are listed at the top, I personally find them awkward when "appendicised" but we can do that if necessary. I plan to add a Table of Contents to the document. My hope was that the "Process summary" section would be the jumping in point if someone doesn't care about any of the other details. Certainly that's where I would jump to if I was reading a similar document.

What do you think?

Reviewer: The ToC should alleviate my concern 👍

Author: Just to clarify on this one, the table of contents is built into the website version of the document so I don't think it's worth duplicating in the text of the document itself. You can see the preview of the website version in the link in the PR description.

The process described below is heavily modified from the regular Sociocratic
process. The differences come from two directions:

> - The needs of a largely asynchronously coordinated base of contributors

Reviewer: ```suggestion
- The needs of an asynchronously coordinated base of contributors
```

- Internal feedback and iteration on the initial process, even after
  modification for asynchrony

> ## When to follow this process
>
> Whether this process is followed for any given proposal or discussion is left
> entirely to the discretion of the people relevant to the discussion. We
> believe that this process helps the project's contributors make expedient and
> clear decisions. However, it should not become an obstacle that is followed
> for every tiny decision. If a decision can efficiently be made without this
> process, contributors are invited to do so.
>
> For the purposes of project planning, however, this process should be the
> default and should be the strong preference, keeping in mind that it can be
> made faster via [round shortcutting](#shortcutting-a-round) and the
> [synchronous lightening process](#optional-synchronous-lightening-process).

## Continuous improvement and exceptions to every rule

**This process is not set in stone.** Nor is it a set of laws that must be
followed at all times. This document uses prescriptive language filled with
imperatives primarily to prevent the process itself from being filled with

> contingencies at every turn. However, as with every other aspect of our ever

Reviewer: > This document uses prescriptive language filled with
imperatives primarily to prevent the process itself from being filled with
contingencies at every turn.

I'm trying to approach this document with accessibility in mind and simplifying language where possible. Might something like this work instead, without losing meaning? This is all very subjective so please don't feel like you _have to_ implement any of these types of suggestions.

> "This document uses prescriptive language to avoid confusion and uncertainty during the process."

Author: In my opinion no, because it's necessary to emphasise the fact that every imperative probably has a qualification (one of the few listed in this section). "This document uses prescriptive language to avoid confusion and uncertainty during the process" doesn't make sense to me as the avoidance of qualifying language throughout the rest of the document isn't to avoid confusion or uncertainty _during_ the process, but to simplify the language of the proposal so that it's actually readable.

Maybe:

> This document uses prescriptive language. However, keep in mind that broad qualifications exist and are described in this section. The document avoids qualifying every imperative in order to prevent the text from ballooning in size and complexity.

What do you think?

Reviewer: Your proposed revision is much clearer to me, great!

Author: Coming in the next commit.

evolving processes, these are merely suggestions that we expect contributors to
adhere to. They cannot cover every situation and undoubtedly there are
exceptional situations where following these suggestions to the letter would be
antithetical to the goals of the project and even of the process itself.

If during your time as an Openverse contributor, you feel there are ways in
which this process can be improved and iterated upon, please know that this
process should be iterated upon and that suggestions for improvement are open

> from any contributor to the project.

Reviewer: Another simplification suggestion, perhaps: `If you have ideas for improving the Openverse decision-making process, please share them. Suggestions are open to any contributor.`

## Terms

- "Author": The person making the proposal.
- "Participants": All participants of the decision-making process, including the
  author, the requested participants, and ad-hoc participants.
- "Reactions": A broad class of feedback that can include personal preferences
  as well as general doubts about the validity or safety of the solutions
  described in the proposal.
- "Question": A request for clarification or explanation of an aspect of the
  proposal. Questions may also request additional information be added to the

>   proposal.

Reviewer 2: Would a suggestion to improve the project be considered a question (as in Can we add .../do it this way?)? One example is this suggestion comment: https://github.com/WordPress/openverse/pull/855#discussion_r1130233362

Author: I would hope so. "Would it make sense to try _x_ approach?" seems like a very reasonable question. Perhaps "Can you clarify why we wouldn't try _x_ instead?", etc, etc.

If the statement ends, grammatically, with an `?`, I presume it is fine. Are there specific types of questions you think we should not expect?

Reviewer 2: Your reply and clarifications make total sense. Thank you for confirmation!

> - "Question": A request for clarification or explanation of an aspect of the
>   proposal. Questions may also request additional information be added to the
>   proposal.

Reviewer: ```suggestion
- "Question": A request for clarification, explanation, or additional information about an aspect of the proposal.
```

- "Objection": An explained uncertainty about the appropriateness, efficacy, or
  safety of a solution
- "Paramount objection": An objection outlining an issue that would cause harm
  to the project or its contributors; especially issues that could cause harm
  and are difficult to reverse and not strictly necessary to accomplish the
  goals of the project

> - "Ratified proposal": A proposal that has undergone the formal decision-making

Reviewer: Would "Approved" work instead of "Ratified" document-wide? If so it might be nice since it overlaps with language already familiar to GitHub users (PR approval, for example).

Author: Sure.

  process and has been accepted as a valid solution for the problem it tries to
  address by the participants of the discussion.
- "Process proposal": A proposal to change processes followed by the
  contributors in the course of stewarding the project towards success.
- "Purpose": The motivation for the proposal. This is typically a statement
  describing the problem the proposal aims to solve. In the original Sociocratic

>   model this is called the "driver".

Reviewer: 👍 "Purpose" works well for our audience, thanks for the change.

> ## Major differences from the original process

Reviewer: This section might also be more useful _after_ explaining our process. I don't think most contributors will have background knowledge of the Sociocratic model so it might be easier to grok the differences after seeing the process outlined.

  I have concerns over"
- "I don't understand the underlying problem, so the solution seems too abstract
  to me"
- "I don't think this solution will solve the problem as described"

Note that these are not questions (they do not ask for clarification) nor are
they objections (they do not point to specific problems with the proposed
solution).

### Removal of initial rounds

Because our process is primarily asynchronous, because our decision-making
happens in public, and because the outcomes of our discussions are ultimately
documented in GitHub or other accessible venues, we can skip some of the initial
rounds. Specifically, the "consent to driver" and the "present the proposal"
rounds cannot work in the same way. Typically, the purpose of a proposal will
have been clarified through a previous discussion. Proposals raised without
prior "buy-in" from other contributors should be done so carefully, with the
understanding that participants may question the purpose and need for the
proposal. The presentation of a proposal happens implicitly when the GitHub PR
or discussion is opened to contain the proposal, so we do not need a special
step for this.

## Process summary

> Full descriptions of each of these rounds of discussion are below. This section

Reviewer: ```suggestion
Full descriptions of each of these rounds of discussion are below. This section
```

Could we start with an explainer that this process breaks conversations into discussion rounds? That alone might be unfamiliar to people.

is meant as a summary for getting an initial overview for people new to the
process or refreshing the broad intentions for folks already familiar with it.

These steps are ordered and must be followed in the order they are presented
here.

> > **Nota bene**: The "goal" column is a very brief summary and is not intended

Reviewer: "Note" instead maybe? Bonus points for the 

```md
> **Note**
> The "goal"column...
```

syntax which renders specially in GitHub flavoured markdown:

> **Note**
> The "goal"column...

| [Objection round](#objection-round)                   | 2 days              | Identify paramount and non-paramount objections                                           |
| [Objection revision round](#objection-revision-round) | Author's discretion | Work with participants to revise the text of the proposal to address paramount objections |
| [Ratification](#ratification)                         | N/A                 | Mark as accepted and create issues to implement the proposal                              |
| [Tabling](#tabling)                                   | N/A                 | Indicate that a proposal will not be implemented                                          |

## Proposal requirements

Project proposals and technical implementation plans should follow the formats
outlined in [the documentation for those processes](./docs/projects/README.md).
All proposals should include a summary of the purpose of the proposal and the
problem(s) it attempts to address.

Every proposal must include the following boilerplate text in the PR description
or at the top of the GitHub Discussion:

```md
## Decision-making process

For the purposes of this discussion, we will follow

> [Openverse’s version of the Consent decision-making model](https://wordpress.github.io/openverse/reference/consent_decision_making.html).

Reviewer: ```suggestion
[Openverse’s decision-making model](https://wordpress.github.io/openverse/reference/consent_decision_making.html).
```

> Please note that this process follows formalised steps with specific

Reviewer: > ~Please note that~ This process follows formalised steps with specific

> expectations of participants. Before jumping in, please read

Reviewer: ```suggestion
expectations of participants. Before contributing, please read
```

[this document](https://wordpress.github.io/openverse/reference/consent_decision_making.html)
as well as
[Openverse’s Code of Conduct](https://github.com/WordPress/openverse/blob/main/CODE_OF_CONDUCT.md).
Please note that the consent decision-making document linked above also includes

> specific provisions for opting out of a decision discussion you do not wish to

Reviewer: ```suggestion
instructions for opting out of a decision discussion you do not wish to
```

or cannot participate in.

## Current round

> The discussion is currently in the {Question, Reaction, Revision, Objection
> raising, Objection revision} round.

Reviewer: ```suggestion
The discussion is currently in the **{Question, Reaction, Revision, Objection
raising, Objection revision}** round.
```

The current round in bold might be nice.

```

## Round descriptions

> Please keep in mind the following general guidelines that apply to all rounds:

Reviewer: ```suggestion
The following general guidelines apply to all rounds:
```

- Reactions may be shared at any of the "feedback" rounds (Question and
  Objection).
- Proposal authors are not obligated to respond to every question, reaction, or
  objection that is raised, except for paramount objections. While there is an
  expectation that the author will address outstanding questions, it isn't
  necessary to respond to every non-paramount objection before moving on to
  subsequent rounds or ratifying the proposal. It is, of course, in the best
  interest of the project and the proposal that as much clarity as possible is
  achieved.
- Authors may table a proposal at any time at their discretion. If the proposal
  is tabled, discussion should not be expected to continue, but may, if there is
  something important being discussed.
- Any round with a suggested length of time is subject to
  [shortcutting as described below](#shortcutting-a-round) or to extension at
  the discretion of the participants.

### Question round

This round lasts for 3 business days after the discussion starts but can be
extended if questions reveal significant gaps in the proposal that must be

> understood before the proposal can be revised in the revision step. These

Reviewer: Simplification: 

> The round lasts for 3 business days but can be extended if important gaps in the proposal are found.

> questions are different from objections in that they’re not in the form of "we

Reviewer: > These questions are different from objections

Can we introduce the goals of this round before clarifying that questions are not objections? Something like "In this round participants ask questions to clarify their understanding of the proposal and identify gaps." or something to that aim.

cannot do this because x part of the proposal will not work". Rather, they are
clarifying questions in the form of:

- "I don’t understand y in the proposal, can you clarify what is meant by this?"
- "I think a gap exists in the proposed solution here [clarify where]. Can you
  fill in the details in this area?"

> In short, the goal of the question round is for everyone participating to fully

Reviewer: > The goal of the question round is for everyone participating to fully
understand the proposal.

This might be a good first sentence for this section.

understand the proposal. Participants are urged that many objections can be
restated as questions. This helps promote good faith between participants and
allows the author to clarify things they might have missed in the initial draft
of the proposal. This does _not_ mean that known paramount objections should be
reserved by participants. If you know that a proposed solution is impossible or
illegal, please do raise this concern as soon as possible.

Note that the author does not have to respond to every question. However, they
should keep them in mind during the revision round.

#### Reactions

> This round also invites participants to share reactions, positive or negative,

Reviewer: > This round ~also~ invites participants to share reactions, positive or negative,

Author: I think the "also" is a fair qualification here considering it is in a sub-heading of the question round :sweat_smile: Without the "also" it reads in context as though "Reactions" is the name of another round.

Reviewer: >  it reads in context as though "Reactions" is the name of another round.

...which was how I read it, oops 😅.

> separate from potential paramount objections and in particular is an excellent
> venue to highlight positive aspects of the proposal regardless of whether
> paramount objections might be raised later. You may also share non-paramount

Reviewer: ```suggestion
separate from potential paramount objections. This is an excellent round
to highlight positive aspects of the proposal, regardless of any paramount 
objections. You may also share non-paramount
```

objections at this round. For example, “I find JavaScript harder to work with
than Python” is an objection but not a paramount objection. What makes an
objection “paramount” is clarified further in the
[Objection round](#objection-round) below.

> Reactions are explicitly expected at this round because, similar to questions,

Reviewer: ```suggestion
Reactions are expected during this round because, similar to questions,
```

they can help prompt further clarification from the author to be address in the
revision round.

#### Purpose

Proposals should all include an explicit "purpose". Often these are informed by
the needs of a particular project associated with the proposal or with a
requested discussion as the result of a retrospective or other self-feedback
mechanism. However, contributors may, at their discretion, raise proposals for
discussion that do not have prior conversations. In these cases, the question
round is also understood to include an evaluation and clarification of the
purpose of the proposal. For regularly requested proposals, however, the purpose

> is taken to be accepted and should not be the focus

Reviewer: Is this in the wrong place? It also ends without a complete sentence.

Author: It was missing a full-stop. Fixed in the next commit.

### Revision round

At this round, the author has the opportunity to revise the proposal in response
to the questions and reactions raised. This round has no set time period. The
author should use their discretion and give an expected time when the revised
version will be ready. The expected date of completion for the revision is both
for accountability (to prevent projects from being stalled indefinitely) and for
clarity of expectations for the other participants in the process. If the author
requires additional time beyond their initial estimation, they should let the
other participants know as soon as possible, keeping in mind that other
discussions may be stalled in the meantime.

> To every extent possible, **no revisions should be made to the proposal before
> this point other than grammatical or spelling corrections**. The reason for this

Reviewer: ```suggestion
**no revisions should be made to the proposal before
this point other than grammatical or spelling corrections**. The reason for this
```

We can avoid clarifications like "To every extent possible," since you explained the imperative language used at the start of the document.

is to allow the question round to fully complete without the proposal changing
during the discussion. This allows clarification to happen fully without having
to track ongoing changes to the proposal and sets the expectation that changes
the proposal incorporate all the clarifying questions asked in addition to any
information surfaced in the reaction round. This helps to minimise the
potentially "frantic" nature of a discussion that is happening about a proposal

> document that is changing simultaneously with the discussion.

Reviewer: Well said. I really think this structure will help us.

### Objection round

This round lasts for 2 business days. Participants are now invited to share
“paramount objections” to the proposal, if any exist. Objections are considered
“paramount” only if:

- They point to an issue that prevents the team from achieving its goals
  (whether these are technical, community, or interpersonal). These issues can
  also be said to "harm" the project or its contributors.
- The objection reveals something that is irreversible (or extremely difficult
  to reverse) of which the participants are not at least “okay” with or “sure
  of”. For example, a proposal to switch to using MySQL when we’re currently
  invested heavily in Postgres would require significant time investment that
  would be difficult to reverse. If the participants are not absolutely certain
  that it is the best decision, then any objection to making the switch would be
  paramount.

Note that during this round, like the question round, the author should not
modify the text of the proposal. The focus of the author in this round should be

> in clarifying objections (especially paramount objections) and deciding whether

Reviewer: ```suggestion
to clarify objections (especially paramount objections) and deciding whether
```

> it will be possible or reasonable to revise the proposal to address them. This

Reviewer: ```suggestion
the proposal can address them. This
```

allows time for participants to fully digest the proposal and raise all
objections without further revision happening. Participants are expected to help
each other decide whether an objection is paramount. Whether an objection is
paramount can be questioned by the author, but they should prefer to defer to

> the participants and trust their judgement.

Reviewer: > they should prefer to defer to the participants and trust their judgement.

This is great.

If there are no paramount objections then the process has finished and the

> proposal is [ratified](#ratification). In that case mark the discussion as

Reviewer 2: Should we use the GitHub tools here: when the reviewer has no paramount objections at this stage, they approve the PR, and if they do have paramount objections, they "request changes"?

ratified, however this can be best done with the tool being used (merge the PR,

> close the discussion, etc). The author is responsible for updating the “Current

Reviewer: From: 

> In that case mark the discussion as ratified, however this can be best done with the tool being used (merge the PR, close the discussion, etc).

To: 

> The proposal author will mark the discussion as ratified by merging their PR or closing their discussion.

round” text to signify the proposal’s ratification.

> At the end of the 2-day period, if the author does not think they can address

Reviewer: ```suggestion
After the 2-day period, if the author does not think they can address
```

> the paramount objections, then the proposal is [tabled](#tabling). For projects
> where we absolutely want to follow through on the idea, this should be rare.

Reviewer: ```suggestion
the paramount objections, then the proposal is [tabled](#tabling). For projects
where we absolutely want to implement, this should be rare.
```

However, sometimes a paramount objection may be raised that requires completely
re-writing the proposal from the ground up. In that case, we will follow this
process again from the start with the new proposal, so we will end the process
for the first proposal at this round and wait for a new one to be raised. In

> these cases, the author should update the “Current round” text to note that was
> tabled and that a new proposal has been requested.

Reviewer: ```suggestion
these cases, the author should update the “Current round” text to note that the proposal was tabled and that a new proposal has been requested.
```

a "good enough" middle ground solution that either completely addresses the
paramount objection or mitigates the potential harm identified.

As stated in the previous round’s description, objection revision should only
happen if paramount objections can indeed be revised to remove the harm. If the
paramount objection involves a core issue of the proposal that cannot be
revised, then an entirely new proposal should be written, rather than the
"revision" replacing the entire existing proposal. It may be the case that in
the process of revision, the author of the proposal realises that they need to
"return to the drawing board". Please keep in mind that this is not a failure
and should be celebrated. So long as the objections were truly paramount, the
participants in the discussion and the author of the proposal have successfully
prevent long-term harm to the goals or people on the project. That is good news.

Once the paramount objections are addressed, we will consider the proposal to be
accepted. The team as a whole can move on the next step of the project, whether
that is implementation planning or implementation itself.

## Outcomes

> There are two possible outcomes for any proposal that follows this process:
> ratification or tabling. All raised proposals should eventually be marked as
> ratified (if the decision-making process warrants it) or tabled.

Reviewer: ```suggestion
There are two possible outcomes for any proposal that follows this process:
ratification or tabling.
```

### Ratification

Ratification occurs once all paramount objections have been addressed and the

> author is satisfied with the state of the proposal. At this point, if there are
> issues to create to implement the proposal, they should be created and linked in
> the proposal text relevant to the issue.

Reviewer: ```suggestion
author is satisfied with the state of the proposal. At this point any necessary GitHub issues should be created and linked to the proposal.
```

## Additional practices

The following additional conventions will help our team make expedient decisions
while still following the formalised process laid out above.

### Discussion Dashboard

The
[discussion dashboard is implemented as a GitHub project](https://github.com/orgs/WordPress/projects/79/views/1)
for tracking the different rounds of a proposal. Proposal authors are expected
to correctly add and update proposals as they move through the different rounds
of discussion. This dashboard is used to
[help ensure we do not assign contributors to more than two discussions in parallel](#minimising-parallel-discussions),
so it is imperative that we maintain its accuracy.

Each discussion should be tracked as a single card, with issues removed from the
"Pending proposal" column if a GitHub PR is opened. If a GitHub discussion is
used for the proposal, because those cannot be represented in GitHub project
boards, the issue should be used to track the status through the columns.

This dashboard also serves as a record of decisions made on the team since the
dashboard's inception. Historical decisions that happened before the dashboard
started being used are not documented there.

> 

Reviewer 2: I've added a couple of views to track everyone's load:
<img width="1088" alt="Screenshot 2023-03-10 at 3 47 54 PM" src="https://user-images.githubusercontent.com/15233243/224320162-74687645-b88b-4b5c-b7d9-aeb6263675ae.png">

I feel that it would be useful, and hope they do not make the board too crowded. What do you think, @sarayourfriend ?

Reviewer 2: Oh, and I added my open proposal :)

#### Column descriptions

For the most part, the columns correspond 1-to-1 with rounds of the Openverse
consent decision-making process. However, the following columns are unique and
bear explaining:

- "Proposal requested": This column is meant for requested proposals or
  discussions for which no formal proposal yet exists. This column should be
  populated by the issues that document the request.
- "Discussion pending": This column is meant for proposals that have been

>   written, but cannot

Reviewer: Missing the rest of the sentence.

   team members. Even in this case, limit to 1 ongoing team-wide process
   discussion per group of contributors. This is less useful if everyone on the
   team wishes to give input in the discussion, although people are free to
   asynchronously give input before the fact.

For both, we will aim to limit the process discussions to one discussion at a
time for any given team member. This is recommended to avoid discussion burn out
as I think I’ve observed process discussions feeling much heavier weight and
burdensome than technical discussions. It also makes sure there is still room
for team members to be involved in one other separate technical discussion and
feel like they can still focus on their other work duties.

### Speeding up decision-making

The base process should help the Openverse project make decisions more quickly
and effectively. However, the following additional practices are available to
help move things along at an even faster clip, if so desired.

#### "Shortcutting" a round

At any point during a round, any participant may say something along the lines
of "no (further) questions", "no reaction", or "no objection" in order to

> short-cut their participation in the round. If all participants are finished

Reviewer 2: Here, too, I think it would be useful to "Approve the PR" after leaving a note like "no objection" to make the merging later easier.

- "Purpose": The motivation for the proposal. This is typically a statement
  describing the problem the proposal aims to solve. In the original Sociocratic
  model this is called the "driver".

## Process summary

The process breaks proposal evaluation and decision-making into a series of
rounds. Each round has expectations about what kinds of feedback are
appropriate. These steps are ordered and must be followed in the order they are
presented here.

Full descriptions for each of these rounds are below. This section is a summary
meant to give an overview for people new to or refreshing themselves on the
process.

> **Note** The "goal" column is a very brief summary and is not intended to
> encapsulate the full scope of the round. Please refer to the
> [round descriptions below](#round-descriptions) for more details and examples.

| round                                                 | Suggested span      | Goal                                                                                      |
| ----------------------------------------------------- | ------------------- | ----------------------------------------------------------------------------------------- |

> | [Question round](#question-round)                     | 3 days              | Clarify the proposal; share initial reactions                                             |

Reviewer 2: What should the process be if there is no response from the maintainers at the end of those 3 (or 5) days? Should we consider that there are no questions, or should we keep the discussion on the current stage until we receive some kind of reply (even if it's "I have no questions") from the assigned reviewers?

| [Revision round](#revision-round)                     | Author's discretion | Update the text of the proposal to reflect the outcome of the previous round              |
| [Objection round](#objection-round)                   | 2 days              | Identify paramount and non-paramount objections                                           |
| [Objection revision round](#objection-revision-round) | Author's discretion | Work with participants to revise the text of the proposal to address paramount objections |
| [Approval](#approval)                                 | N/A                 | Mark as approved and create issues to implement the proposal                              |

> | [Tabling](#tabling)                                   | N/A                 | Indicate that a proposal will not be implemented                                          |

Author: Yup! I didn't include it because I wasn't sure if MyST supported mermaid diagrams but it does (https://myst-tools.org/docs/mystjs/diagrams) so I'll add it during revision.

## Proposal requirements

Project proposals and technical implementation plans should follow the formats
outlined in [the documentation for those processes](./docs/projects/README.md).
All proposals should include a summary of the purpose of the proposal and the
problem(s) it attempts to address.

Every proposal must include the following boilerplate text in the PR description
or at the top of the GitHub Discussion:

```md
## Decision-making process

For the purposes of this discussion, we will follow
[Openverse's decision-making model](https://wordpress.github.io/openverse/reference/consent_decision_making.html).
This process follows formalised steps with specific expectations of

> participants. Before contributing, please read

Reviewer 2: I am trying to word my suggestion as a question:)

Could we not make it sound as if following the process is a hard requirement? 

It seems a very large ask for the community to read and understand a long document before adding a short comment. For instance, we might want some feedback from the designers or want ideas from the larger community on devex. I think having the requirements written in such a way might push non-maintainers away from contributing.

Having a clear indication of the current stage and the links to project documentation is very useful for maintainers. Reminding to read the docs shouldn't be necessary for the maintainers and might be off-putting for outside contributors.

Author: For sure. Zack brought a similar concern up that it's a long document to ask someone who is making an occasional consultation-style contribution to a discussion to read and understand.

I'd like to add something to qualify that, basically to carve out an exception for folks to be able to participate in that way when necessary. I do think it's useful to have the expectation that the process will be followed, but I also think we're capable of managing it on our own and deciding whether we're getting sidetracked from the core discussion.

> entirely to the discretion of the people relevant to the discussion. We
> believe that this process helps the project's contributors make expedient and
> clear decisions. However, it should not become an obstacle that is followed
> for every tiny decision. If a decision can efficiently be made without this
> process, contributors are invited to do so.
>
> For the purposes of project planning, however, this process should be the
> default and should be the strong preference, keeping in mind that it can be
> made faster via [round shortcutting](#shortcutting-a-round) and the
> [synchronous lightening process](#optional-synchronous-lightening-process).

## Continuous improvement and exceptions to every rule

**This process is not set in stone.** Nor is it a set of laws that must be
followed at all times. This document uses prescriptive language. However, keep
in mind that broad qualifications exist and are described in this section. The
document avoids qualifying every imperative in order to prevent the text from
ballooning in size and complexity.

As with every other aspect of our ever evolving processes, this process is

> merely a guideline that we expect contributors to adhere to. It cannot cover

Author: I like the language you've used, thanks for the suggestion. I'll make this change during revision.

- "Purpose": The motivation for the proposal. This is typically a statement
  describing the problem the proposal aims to solve. In the original Sociocratic
  model this is called the "driver".

## Process summary

The process breaks proposal evaluation and decision-making into a series of
rounds. Each round has expectations about what kinds of feedback are
appropriate. These steps are ordered and must be followed in the order they are
presented here.

Full descriptions for each of these rounds are below. This section is a summary
meant to give an overview for people new to or refreshing themselves on the
process.

> **Note** The "goal" column is a very brief summary and is not intended to
> encapsulate the full scope of the round. Please refer to the
> [round descriptions below](#round-descriptions) for more details and examples.

| round                                                 | Suggested span      | Goal                                                                                      |
| ----------------------------------------------------- | ------------------- | ----------------------------------------------------------------------------------------- |

> | [Question round](#question-round)                     | 3 days              | Clarify the proposal; share initial reactions                                             |

Author: I agree it needs a name change. I've suggested another in another comment to go along with the wider discussion about the scope of this round ("Clarification").

> | [Question round](#question-round)                     | 3 days              | Clarify the proposal; share initial reactions                                             |

Author: I don't have a strong preference for a specific number of days. I err on the shorter side personally because of the following reasons:

1. Decision-making and discussion are part of our regular work. The only PR review priority that is 5 days are low priority PRs. 3 days sits between high and medium in this scale. Considering many discussions block further work, 3 days seems like an appropriate amount of time.
2. This is a suggested span and can be expanded by the author if, for example, the proposal is very long, or if the participants request it. It's also not a "hard" deadline in that (as specified in the round description) if there are still important outstanding questions, the round should continue.

Anything less than a week still brings the sense of urgency and expediency that I hope this will help us capture in our decision-making. 5 days sounds fine to me in that sense, but I still think we can consider it in light of the ideas I shared above.

I also think this proposal qualifies for "very long" and I anticipate it going over 3 days in the initial round. On the other hand, the #873 review did not take longer than 5 days total once the requested reviewers took a look at it. If we go for a longer period of time, I'd hope we can more proactively use the shortcutting signal instead of letting a round expire on its own.

> | [Question round](#question-round)                     | 3 days              | Clarify the proposal; share initial reactions                                             |

Reviewer 3: > I'd hope we can more proactively use the shortcutting signal instead of letting a round expire on its own.

💯 I love all the levers we have here for shortening/lengthening the spans as necessary -- this process is going to be useful for such a wide range of discussions that there's no way to make a default that will make sense in _all_ cases.

The only pro I see for 5 is that it makes overlap with a community meeting part of the _default_, rather than something an author might have to go out of their way to do. But that's such a specific detail, I'm 100% down to just wait and see what the time spans end up looking like once we roll out the process.

| [Revision round](#revision-round)                     | Author's discretion | Update the text of the proposal to reflect the outcome of the previous round              |

> | [Objection round](#objection-round)                   | 2 days              | Identify paramount and non-paramount objections                                           |

Author: As in the other round name comment, I'll roll this discussion into the wider discussion of clarifying the role of the first round.

  also be said to "harm" the project or its contributors.
- The objection reveals something that is irreversible (or extremely difficult
  to reverse) of which the participants are not at least "okay" with or "sure
  of". For example, a proposal to switch to using MySQL when we're currently
  invested heavily in Postgres would require significant time investment that
  would be difficult to reverse. If the participants are not absolutely certain
  that it is the best decision, then any objection to making the switch would be
  paramount.

During this round, as in the question round, the author should not modify the
text of the proposal. The focus of the author in this round should be to clarify
objections (especially paramount objections) and decide whether the proposal can
address them. This allows time for participants to fully digest the proposal and
raise all objections without further revision happening. Participants are
expected to help each other decide whether an objection is paramount. Whether an
objection is paramount can be questioned by the author, but they should prefer
to defer to the participants and trust their judgement.

> If there are no paramount objections then the process has finished and the

Author: I think I'd rather keep them in the objection round description. I think that's where it's most natural to look for how to decide which round to go to next, but I could be wrong about that. In any case, point taken about the duplication and one way or another I will fix this during revision.

accepted. The team as a whole can move on the next step of the project, whether
that is implementation planning or implementation itself.

## Outcomes

There are two possible outcomes for any proposal that follows this process:
approval or tabling. In both cases, the author is expected to:

- Update the current round callout to indicate the outcome
- If the proposal is in a GitHub discussion, lock the discussion
- If the proposal is in a PR, merge the PR if approved or close it if tabled
- Move the associated card on the discussion dashboard to the appropriate column

Further requirements specific to each outcome follow.

### Approval

Approval occurs once all paramount objections have been addressed and the author
is satisfied with the state of the proposal. At this point any necessary GitHub
issues should be created and linked to the proposal.

### Tabling

Tabling can occur in the following cases:

- At any time, at the discretion of the author
- Before a proposal is written, if the team reneges on a requested discussion
- After a proposal is written, but before the discussion happens, if it is clear
  the proposal will not fit the needs of the project

> - During the objection round when insurmountable paramount objections are raised

Author: I'm not sure I understand the question.

If you're asking whether the fact that the author can table the discussion at their discretion at any time, I think it so. If someone no longer believes in their proposal I don't think it makes sense to ask them to continue to clarify, revise, and respond to objections with respect to it. If another person wishes to take it up they can base their proposal on the existing text and start a new decision-making process for it.

If you're asking whether all four points can be summarised as "at the author's discretion", then I think it's _technically_ true. But I also think clarifying specific, expected situations where tabling would occur is warranted. Not all of these points are necessarily initiated by the author. It could be that an external stakeholder decides a project is no longer necessary, causing the third case to happen. I wouldn't say that's the "author's discretion" but is a realistic and plausible case that could arise.

Before requesting participation from specific people on a discussion, check the
[discussion dashboard](#discussion-dashboard) to ensure they're not already
involved in 2 ongoing discussions. If specific participants are required for a
given proposal (due to the relevance of their expertise in the proposal), then
you may open the proposal, but it should be left in the "pending discussion"
column.

Tips for checking the dashboard and not accidentally over-encumbering people:

- Check for discussions in the "pending discussion" column. If someone is
  already assigned to discussions there, then know that any new discussions that
  _must_ involve them will need to be triaged with the other discussion
- If a discussion could happen using the "lightening process", then you may be
  able to pull people in who are waiting for a discussion to restart during the
  revision round, even if they're already participating in the discussion

In either case, it is still courteous to check with the individuals and make it
clear that they can push the discussion to someone else (provided they're not a
necessary subject-matter expert).

#### Project level process discussions

> 

Reviewer 3: Pre-emptively resolving this as. something that can be addressed later.

particularly helpful for simpler discussions where the 2 business day
recommended time periods can feel overly slow.

#### Optional, synchronous "lightening process"

If all the participants of a particular discussion are able to participate in a
synchronous version of the process outlined above, they may elect to do so. This
can be particularly useful for relatively small but impactful decisions where
the 2 business day periods would unnecessarily drag the process out. In these
cases, participants should hold a synchronous conversation with the
understanding that someone will take notes for each round, recording the
questions, reactions, and any objections that arise. **The notes should be
shared afterwards in the original proposal venue and should include explicit
headings per round.** If necessary, the synchronous discussion can be broken
into two separate sections to allow time for the revision round after reactions
are shared.

Even in these cases, however, the text of the proposal should still be written
in a GitHub PR or discussion and available for all participants to read _before_
the lightening process.

> Differences in structure for the "lightening process" from the asynchronous

Author: The documentation of the full process would still require changes. We don't do the "consent to driver" step. We also call the "driver" the purpose. If we link to the original process documentation then we should also be cognisant of the fact that we're asking people to read another long description of the process. With the summary here, it allows people instead to just adapt the process they already know to a synchronous one. Considering the tweaks are small (two points), the vocabulary differences, and the differences in initial process, I think the summary in this document is more useful.

> Differences in structure for the "lightening process" from the asynchronous

Reviewer 3: Oh, I had totally forgotten about that difference -- which proves the point that it _is_ useful to have this summary! 😄 Makes sense to me :)

    the proposal can be left as inline comments attached to a review requesting
    changes.
- General PR comments
  - For questions or non-paramount objections raised regarding the proposal in
    general or for the reaction round.
  - In addition to the PR description, the author of the PR will also create a
    new top-level PR comment when a new round starts.
- Review comments
  - These should be used in the objection raising round. Leave an approving
    review if you have no paramount objections or "request changes" if you have
    a paramount objection. Attach inline comments as needed in either case.

### GitHub Discussions

Because the format is slightly more limited than PRs, we will have a top-level
comment per round created by the author. Responses from participants should be
contained within the thread for each round. However, paramount objections during
the Objection round, should be raised as individual threads. Still, the author
should still create a top-level comment closing the previous round and opening
the objection round. If the proposal goes into objection revision, the
discussion for addressing a given objection can happen in the accompanying
thread.

> ## Major differences from the original process

Reviewer 3: I'm not sure whether to pre-emptively resolve this so will leave it up. The part about potentially moving this into a separate document is definitely 'formatting' work which, to reiterate, we definitely don't need to address now.

Regardless of where the documentation lives, I think the point about it being difficult to explain the differences without explaining the original method stands. I'm leaving this as a _suggestion_ that removing it (temporarily, if folks think it would be valuable to add back in) and not trying to directly compare our process to the quite-different inspiration would be the minimal effort approach for a v1 doc.
